src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.cs

```diff
@@ -17,8 +17,24 @@ internal static partial class Crypto
         [return: MarshalAs(UnmanagedType.Bool)]
         private static extern bool EcDsaSign(ref byte dgst, int dlen, ref byte sig, [In, Out] ref int siglen, SafeEcKeyHandle ecKey);
 
-        internal static unsafe int EcDsaVerify(ReadOnlySpan<byte> dgst, int dgst_len, ReadOnlySpan<byte> sigbuf, int sig_len, SafeEcKeyHandle ecKey) =>
-            EcDsaVerify(ref MemoryMarshal.GetReference(dgst), dgst_len, ref MemoryMarshal.GetReference(sigbuf), sig_len, ecKey);
+        internal static int EcDsaVerify(ReadOnlySpan<byte> dgst, ReadOnlySpan<byte> sigbuf, SafeEcKeyHandle ecKey)
+        {
+            int ret = EcDsaVerify(
+                ref MemoryMarshal.GetReference(dgst),
+                dgst.Length,
+                ref MemoryMarshal.GetReference(sigbuf),
+                sigbuf.Length,
+                ecKey);
+
+            if (ret == 0)
+            {
+                // OpenSSL ECDSA signature processing requires a DER encode and decode, so
+                // the error queue may have been contaminated.
+                ErrClearError();
+            }
+
+            return ret;
+        }
```

Comment:
@pjanotti, @bartonjs, it looks like EcDsaVerify calls to:
https://github.com/dotnet/corefx/blob/77732da1052923259a7668540ee62bf48d35956e/src/Native/Unix/System.Security.Cryptography.Native/pal_ecdsa.cpp#L23-L26
which just calls ECDSA_verify, which according to https://linux.die.net/man/3/ecdsa states "ECDSA_verify() and ECDSA_do_verify() return 1 for a valid signature, 0 for an invalid signature and -1 on error."  But we're only checking for `ret == 0` here.  Should that be `ret <= 0`?  Or have I misread the code or docs?

Reply:
I just double-checked, and this should only end up touching the queue in cases it returns negative.  (Hardware module interactions can always break these assumptions, but for software keys it should be only `< 0` cases which carry the need to clear (as opposed to the `== 0` that I wrote... good catch).

Reply:
No, you're right. I will update it.